packages/ui5-application-writer/src/options.ts

```diff
@@ -56,7 +57,11 @@ const factories: { [key: string]: (input: FeatureInput) => Promise<void> } = {
     codeAssist: async (input: FeatureInput) => await copyTemplates('codeAssist', input),
     eslint: async (input: FeatureInput) => await copyTemplates('eslint', input),
     loadReuseLibs: async (input: FeatureInput) => await copyTemplates('loadReuseLibs', input),
-    sapux: async (input: FeatureInput) => await copyTemplates('sapux', input),
+    sapux: async (input: FeatureInput) => {
+        if (input.ui5App.app.projectType === 'EDMXBackend') {
```

Comment:
Do you need to apply some optional chaining check here for undefined? Its a deeply nested object, will these values always exist?

Reply:
> Do you need to apply some optional chaining check here for undefined? Its a deeply nested object, will these values always exist?

`ui5App` and `app` are mandatory properties of [FeatureInput](https://github.com/SAP/open-ux-tools/pull/2147/files#diff-8651a8bccdacbc50e0fc8b935e886f22d73e6ad4a7b3f02da31c28b740bafa7dR15). `ProjectType` within `app` is optional.  If `projectType` is undefined, the condition `input.ui5App.app.projectType === 'EDMXBackend'` will safely evaluate to false know? so do we need optional chaining here? Also `projectType` is mandatory for `Ui5App`, so `projectType` will always be provided at this point. What do you think ?